### PR TITLE
fixes #16824 - Change repo upstream uname/password

### PR DESCRIPTION
**Pull request by parthaa**
This commit enables one to properly  update and reset
the upstream username and password information for a
repository

### CODE REVIEW

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/repository-details-info.controller.js [hunk 1]
```diff
@@ -114,5 +114,10 @@ angular.module('Bastion.repositories').controller('RepositoryDetailsInfoControll
             }
             return OstreeUpstreamSyncPolicy.syncPolicyName(policy);
         };
+
+        $scope.clearUpstreamPassword = function () {
```
Comment (waldenraines) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/repository-details-info.controller.js]: Please add a test for this.

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/repository-details-info.filter.js [hunk 2]
```diff
@@ -18,3 +18,12 @@ angular.module('Bastion.components.formatters').filter('ostreeUpstreamSyncPolicy
         return OstreeUpstreamSyncPolicy.syncPolicyName(policy);
     };
 }]);
+
+angular.module('Bastion.components.formatters').filter('upstreamPasswordFilter', [function () {
```
Comment (waldenraines) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/repository-details-info.filter.js]: Please add a test for this.

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/repository-details-info.controller.js [hunk 3]
```diff
@@ -114,5 +114,10 @@ angular.module('Bastion.repositories').controller('RepositoryDetailsInfoControll
...
+            $scope.repository['upstream_password'] = '';
```
Comment (waldenraines) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/repository-details-info.controller.js]: Wonder if you should set this to null instead?

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/repository-details-info.filter.js [hunk 4]
```diff
@@ -18,3 +18,12 @@ angular.module('Bastion.components.formatters').filter('ostreeUpstreamSyncPolicy
...
+    return function (displayValue, repository) {
+        if (repository["upstream_password_exists"]) {
+            return "******";
+        }
+        return "";
```
Comment (waldenraines) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/repository-details-info.filter.js]: What about returning null instead?

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/repository-details-info.filter.js (hunk 2 above)
Comment (parthaa) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/repository-details-info.filter.js]: @waldenraines added

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/repository-details-info.controller.js (hunk 1 above)
Comment (parthaa) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/repository-details-info.controller.js]: @waldenraines added tests

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/views/repository-info.html [hunk 5]
```diff
@@ -49,9 +49,17 @@ <h4 translate>Basic Information</h4>
...
+          readonly="denied('edit_products', product)"
           on-save="save(repository)"
-          readonly="product.redhat || denied('edit_products', product)">
+          formatter="upstreamPasswordFilter"
+          formatter-options="repository"
+          deletable="repository.upstream_password_exists"
+          on-delete="clearUpstreamPassword()"
+          >
+          <input id="upstream_password"
+            name="upstream_password"
```
Comment (waldenraines) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/views/repository-info.html]: Shouldn't this input be of type password?  I can view the password as I type it.

**Comment by waldenraines** [comment]
I'm getting a JS error that is unrelated to your PR but I think is preventing me from testing this fully.  Let me fix real quick (http://projects.theforeman.org/issues/19178).

**Comment by waldenraines** [comment]
Requires https://github.com/Katello/katello/pull/6732

**Comment by waldenraines** [comment]
Everything I tested worked for me, just have that one comment above.

**Comment by waldenraines** [comment]
One other question: is it possible to have a username and not a password or a password without a username?  You can currently do both in the UI.

**Comment by parthaa** [comment]
> One other question: is it possible to have a username and not a password or a password without a username? You can currently do both in the UI.

Unless we allow changing both username/passwd at the same time I dont think we should enforce this.

**Comment by waldenraines** [comment]
> Unless we allow changing both username/passwd at the same time I dont think we should enforce this.

Good point.

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/views/repository-info.html (hunk 5 above)
Comment (parthaa) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/views/repository-info.html]: @waldenraines updated

**Comment by waldenraines** [comment]
> LGTM

But still probably don't want to merge this until #6732 goes in.

**Pull request merged**